provider/google.go

```diff
@@ -284,9 +322,17 @@ func newRecords(endpoints []*endpoint.Endpoint) []*dns.ResourceRecordSet {
 
 // newRecord returns a RecordSet based on the given endpoint.
 func newRecord(endpoint *endpoint.Endpoint) *dns.ResourceRecordSet {
+	// TODO(linki): works around appending a trailing dot to TXT records. I think
+	// we should go back to storing DNS names with a trailing dot internally. This
+	// way we can use it has is here and trim it off if it exists when necessary.
+	target := endpoint.Target
+	if suitableType(endpoint) == "CNAME" {
+		target = ensureTrailingDot(target)
```

Comment:
IMO `ensureTrailingDot` should be moved to `provider.go` file and all the logic regarding adding a dot or not adding should be placed there.